internal/controller/storageclient_controller.go

```diff
@@ -239,6 +241,23 @@ func (r *storageClientReconcile) reconcile(ctx context.Context, req ctrl.Request
 
 	result, reconcileErr := r.reconcilePhases()
 
+	storageClaimList := &metav1.PartialObjectMetadataList{}
+	storageClaimList.SetGroupVersionKind(v1alpha1.GroupVersion.WithKind("StorageClaimList"))
+	if err := r.list(storageClaimList); err != nil {
+		r.log.Info("No StorageClaims remainings")
```

Comment:
we can't deduce a failed api call on a list as no items present and here you should requeue w/ an error.